lib/vmdb/plugins.rb

```diff
@@ -80,6 +86,48 @@ def register_models
 
     private
 
+    # Determine the version of the specified engine
+    #
+    # If the gem is
+    # - git based, pointing to a branch:   <branch>@<sha>
+    # - git based, pointing to a tag:      <tag>@<sha>
+    # - git based, pointing to a sha:      <sha>
+    # - path based, with git, on a branch: <branch>@<sha>
+    # - path based, with git, on a tag:    <tag>@<sha>
+    # - path based, with git, on a sha:    <sha>
+    # - path based, without git:           nil
+    # - a real gem:                        <gem_version>
+    def version(engine)
+      spec = bundler_specs_by_path[engine.root.to_s]
+
+      case spec.source
+      when Bundler::Source::Git
+        [
+          spec.source.branch || spec.source.options["tag"],
+          spec.source.revision.presence[0, 8]
+        ].compact.join("@").presence
+      when Bundler::Source::Path
+        if engine.root.join(".git").exist?
+          branch = sha = nil
+          Dir.chdir(engine.root) do
+            branch   = `git rev-parse --abbrev-ref HEAD 2>/dev/null`.strip.presence
```

Comment:
Not that this (or the next three lines) is wrong, but is is possible to call out to bundler for some of this so we aren't doing what seems like the same work that bundler already does again?

It would probably require re-wrapping the bundler spec in another one, but might save duplicating some work with the shelled out methods.  Just a thought.